src/app/shared/filesystem/file-preview/file-preview.component.ts

```diff
@@ -1,15 +1,23 @@
 import { Component, OnInit } from '@angular/core';
+import { AuthService } from 'app/core/auth.service';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
 
 @Component({
   selector: 'clark-file-preview',
   templateUrl: './file-preview.component.html',
   styleUrls: ['./file-preview.component.scss']
 })
 export class FilePreviewComponent implements OnInit {
+  private isDestroyed$ = new Subject<void>();
```

Comment:
Looks like the `isDestroyed` pattern has started being used here. We should be sure to add ngOnDestroy in the following way to clean everything up:

```
ngOnDestroy() {
  this.isDestroyed$.next();
  this.isDestroyed$.unsubscribe();
}
```